### PR TITLE
fix for driver.scaling_report

### PR DESCRIPTION
**Pull request by naylor-b**
### Summary

Problem was happening when scaling_report was called after driver totals had already been computed in some format other than 'array' format.

Also found and fixed a coloring bug where parts of the jacobian were ending up nonzero when they should have been zero based on the coloring.

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: openmdao/core/total_jac.py [hunk 1]
```diff
@@ -950,10 +950,7 @@ def simul_coloring_iter(self, imeta, mode):
 
         for color, ilist in enumerate(coloring.color_iter(mode)):
             if len(ilist) == 1:
-                if both:
```
Comment (Kenneth-T-Moore) [inline openmdao/core/total_jac.py:953]: I think you can get rid of the variable `both` now.

**Review by Kenneth-T-Moore — CHANGES_REQUESTED** [review]

File: openmdao/core/total_jac.py (hunk 1 above)
Comment (naylor-b) [reply openmdao/core/total_jac.py:953]: fixed

File: openmdao/visualization/scaling_viewer/scaling_table.html [hunk 2]
```diff
@@ -466,31 +472,31 @@ <h3>Jacobian Info</h3>
...
-        svg.append("g")
-            .attr("transform", "translate(0," + height + ")")
-            .call(d3.axisBottom(x))
-            .selectAll("text")
-            .style("text-anchor", "end")
-            .attr("transform", "rotate(-45)")
-            .attr("dx", "-.5em")
-            .attr("dy", ".3em");
-    }
+    // if (is_top) {
```
Comment (robfalck) [inline openmdao/visualization/scaling_viewer/scaling_table.html:475]: Is there a reason to leave these commented-out instead of just removing them?

Comment (naylor-b) [reply openmdao/visualization/scaling_viewer/scaling_table.html]: Nope.  Removed.

**Review by swryan — APPROVED** [review]

**Review by Kenneth-T-Moore — APPROVED** [review]

**Review by robfalck — APPROVED** [review]

**Pull request merged**